File: server/src/models/importData.js
/**
 * Parse the json file from ../../../US_recipes_null.json and store in database
 */

const fs = require('fs')
const path = require('path')
const mongoose = require('mongoose')
const Recipe = require('./recipeModel')

const filePath = path.join(__dirname, '../../../US_recipes_null.json')
const fileData = fs.readFileSync(filePath, 'utf-8')
const recipes = JSON.parse(fileData)

mongoose.connect('mongodb+srv://admin:admin@cluster0.toiwrnt.mongodb.net/').then(()=>{
    console.log('db connected')
})

async function importData() {
  try {
    const recipesArray = Object.values(recipes);
    if (recipesArray.length === 0) {
        console.log("No recipes found in the JSON file.");
        return;
    }
    const result = await Recipe.insertMany(recipesArray);
    console.log(`${result.insertedCount} documents were successfully inserted.`);
  } catch (err) {
    console.error("An error occurred:", err);
  }
}

importData()

File: backend/models/importData.py
import os
import json
from pymongo import MongoClient

with open(file_path, 'r', encoding='utf-8') as f:
    recipes = json.load(f)

client = MongoClient('mongodb+srv://admin:admin@cluster0.toiwrnt.mongodb.net/')
db = client.get_database()
recipe_collection = db['recipes']

def import_data():
    recipes_array = list(recipes.values())
    if not recipes_array:
        print("No recipes found in the JSON file.")
        return
    result = recipe_collection.insert_many(recipes_array)
    print(f"{len(result.inserted_ids)} documents were successfully inserted.")

import_data()